src/main/java/jdraw/std/StdContext.java

```diff
@@ -302,10 +315,59 @@ public boolean accept(File f) {
         int res = chooser.showOpenDialog(this);
 
         if (res == JFileChooser.APPROVE_OPTION) {
-            // read jdraw graphic
-            System.out.println("read file "
-                    + chooser.getSelectedFile().getName());
+            var path = chooser.getSelectedFile().getAbsolutePath();
+            try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(path))) {
+                getModel().removeAllFigures();
+                var figures = (List<Figure>) ois.readObject();
+                for (var f : figures) {
+                    getView().getModel().addFigure(f);
+                }
+            } catch (Exception ex) {
+                throw new RuntimeException((ex.getMessage()));
+            }
         }
     }
 
+    private List<Figure> clipboard;
+
+    private void cut(DrawView view) {
+        if (view.getSelection().size() > 0) {
+            clipboard = view.getSelection();
+            var model = view.getModel();
+            view.getSelection().stream().forEach(f -> {
+                        view.removeFromSelection(f);
```

Comment:
I think this is not necessary as removing a figure from the model removes it also from the selection.

Reply:
:100: :+1: